module/VuFind/src/VuFind/AjaxHandler/GetItemStatuses.php

```diff
@@ -473,65 +468,36 @@ protected function getItemStatusError($record, $msg = '')
         ];
     }
 
-    /**
-     * Convert availability boolean or int to a string
-     *
-     * @param mixed $available Boolean value or one of the ItemStatus::STATUS_*
-     *                         constants
-     *
-     * @return string
-     */
-    protected function availabilityToString($available): string
-    {
-        switch ((int)$available) {
-            case ItemStatus::STATUS_UNAVAILABLE:
-                return 'false';
-            case ItemStatus::STATUS_UNCERTAIN:
-                return 'uncertain';
-            default:
-                return 'true';
-        }
-    }
-
     /**
      * Get a message for availability status
      *
-     * @param array  $messages         Available messages
-     * @param string $available        String 'false', 'true' or 'uncertain'
-     * @param bool   $useUnknownStatus Whether to use the "unknown status" message
+     * @param AvailabilityStatusInterface $availability Availability Status
      *
      * @return string
      */
-    protected function getAvailabilityMessage(
-        array $messages,
-        $available,
-        bool $useUnknownStatus
-    ): string {
-        if ($useUnknownStatus) {
-            $key = 'unknown';
-        } elseif ('false' === $available) {
-            $key = 'unavailable';
-        } elseif ('uncertain' === $available) {
-            $key = 'uncertain';
-        } else {
-            $key = 'available';
-        }
-        return $messages[$key] ?? '';
+    protected function getAvailabilityMessage($availability): string
```

Comment:
You could probably put a typehint on the argument here since the method is being totally revised anyway.

Reply:
Done